README.md

```diff
@@ -13,3 +15,25 @@
   complexity suggests a different time.
 
 Add your answers to this markdown file.
+
+## Solution
+
+**_List 3 reasons why asymptotic analysis may be misleading with respect to actual performance in practice._**
+
+1. Asymptotic analysis focuses on very large input sizes. This means input sizes which approach infinity, but in practice input sizes can be much smaller in alot of situations. An algorithm with better asymptotic complexity may have larger overhead that makes it perform worse for smaller input sizes. This is clear when comparing the graph of diferent time complexities. A polynomial or linear algorithm can perform better with a smaller input size then its logarithmic counterpart. Then as the input gets larger it will change and now the logarithmic complexity will be best.
```

Comment:
But we've seen examples of algorithms that behave according to their asymptotic complexity from small input sizes.

Reply:
Yes. But some don't. EG a linear vs logarithmic with a small input. A linear complexity
![Screenshot 2024-04-20 at 1 51 40 PM](https://github.com/COSC3020/theory-vs-practice-tlaceby/assets/31893030/f143620f-2b1b-4d51-aeb2-5e4aecd0c985)

Here is above an example of just this. In reality with small inputs logarithmic algorithms could behave worse than linear or even polynomial.

Reply:
But it is certainly not true that "Asymptotic analysis focuses on very large input sizes."